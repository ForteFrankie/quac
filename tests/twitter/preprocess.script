#!/bin/bash
# Copyright (c) Los Alamos National Security, LLC, and others.

# Make sure the Twitter preprocessing works.

. ./environment.sh

cd $DATADIR
exec 2>&1

parsesh=$QUACBASE/misc/parse.sh

mkdir raw pre geo raw/foo
ln -s $TWEETDIR/raw/small/small.json.gz raw/foo
touch raw/foo/small.stats
touch -d19700101 raw/foo/small.json.d  # suppress "No such file or directory"

# lots of order variability in this command, so ignore the output
z $parsesh 1 . > /dev/null

x ls -R .

y 'wc -l pre/*.tsv'
y 'md5sum pre/*.tsv'
x pickle-dump pre/metadata
