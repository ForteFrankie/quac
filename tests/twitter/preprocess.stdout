$ ls -R .
.:
geo
pre
raw

./geo:

./pre:
2012-10-28.all.tsv
2012-10-28.geo.tsv
2012-10-29.all.tsv
2012-10-29.geo.tsv
2012-10-30.all.tsv
2012-10-30.geo.tsv
2012-10-31.all.tsv
2012-10-31.geo.tsv
2012-11-01.all.tsv
2012-11-01.geo.tsv
2012-11-02.all.tsv
2012-11-02.geo.tsv
2012-11-03.all.tsv
2012-11-03.geo.tsv
2012-11-04.all.tsv
2012-11-04.geo.tsv
2012-11-05.all.tsv
2012-11-05.geo.tsv
gaps.gp.pdf
geotag_rate.gp.pdf
metadata
summary.tsv
tweet_ct.gp.pdf

./raw:
foo

./raw/foo:
small.2012-10-28.raw.tsv
small.2012-10-29.raw.tsv
small.2012-10-30.raw.tsv
small.2012-10-31.raw.tsv
small.2012-11-01.raw.tsv
small.2012-11-02.raw.tsv
small.2012-11-03.raw.tsv
small.2012-11-04.raw.tsv
small.2012-11-05.raw.tsv
small.json.d
small.json.gz
small.stats
$ (wc -l pre/*.tsv)
   102 pre/2012-10-28.all.tsv
    89 pre/2012-10-28.geo.tsv
   158 pre/2012-10-29.all.tsv
   134 pre/2012-10-29.geo.tsv
   157 pre/2012-10-30.all.tsv
   135 pre/2012-10-30.geo.tsv
   119 pre/2012-10-31.all.tsv
   108 pre/2012-10-31.geo.tsv
   144 pre/2012-11-01.all.tsv
   135 pre/2012-11-01.geo.tsv
   128 pre/2012-11-02.all.tsv
   114 pre/2012-11-02.geo.tsv
   142 pre/2012-11-03.all.tsv
   123 pre/2012-11-03.geo.tsv
   145 pre/2012-11-04.all.tsv
   120 pre/2012-11-04.geo.tsv
    70 pre/2012-11-05.all.tsv
    59 pre/2012-11-05.geo.tsv
     9 pre/summary.tsv
  2191 total
$ (md5sum pre/*.tsv)
09e4490cdbeb2ca7aa7ed9ae73820d47  pre/2012-10-28.all.tsv
395b06b6fbf486900c5533a62a3d2820  pre/2012-10-28.geo.tsv
1aeb8249f0ea7b872401b3ba65d9f0d7  pre/2012-10-29.all.tsv
eadcab7158fa4665f58db5569fd7be5c  pre/2012-10-29.geo.tsv
2341e5921d1c63066e5d0e19028cc20e  pre/2012-10-30.all.tsv
fd728343ff9edd44c9e6be7a9293c7c2  pre/2012-10-30.geo.tsv
934740aafbbbf60bec5e9ddd209ddbf8  pre/2012-10-31.all.tsv
a48937f2579c11688dda9ed314de95d1  pre/2012-10-31.geo.tsv
528d113dd848ded52b4faa025c19771c  pre/2012-11-01.all.tsv
9e62d5198d158c45fa6460cf63f85f07  pre/2012-11-01.geo.tsv
5dc07db0dd0761ca21cbdbb736b6daf0  pre/2012-11-02.all.tsv
4705997bca3ccb6159a325a53aa3f124  pre/2012-11-02.geo.tsv
aff978531ff0b2569fe9176a6ef6b62b  pre/2012-11-03.all.tsv
da2f250bb6fd9f1abed8e0a1c119f2e0  pre/2012-11-03.geo.tsv
c75787bb7d6386f273f3f3731ed5517a  pre/2012-11-04.all.tsv
161687a3a4fb67a150db39c1cb6ddd6b  pre/2012-11-04.geo.tsv
fd6dcf3d69292abb0d7c05563eca18aa  pre/2012-11-05.all.tsv
8a9f29e08badd2f8f09a8b5355369a2f  pre/2012-11-05.geo.tsv
ef397f1b339718a8c323bf13757cb5f8  pre/summary.tsv
$ pickle-dump pre/metadata
{'days': {datetime.date(2012, 10, 28): {'count': 102,
                                        'count_geotag': 89,
                                        'max_id': 262705164329287680,
                                        'min_id': 262441476850581504},
          datetime.date(2012, 10, 29): {'count': 158,
                                        'count_geotag': 134,
                                        'max_id': 263059826312085504,
                                        'min_id': 262707005641347073},
          datetime.date(2012, 10, 30): {'count': 157,
                                        'count_geotag': 135,
                                        'max_id': 263429919097384960,
                                        'min_id': 263067950678958080},
          datetime.date(2012, 10, 31): {'count': 119,
                                        'count_geotag': 108,
                                        'max_id': 263784102887567360,
                                        'min_id': 263430544036098048},
          datetime.date(2012, 11, 1): {'count': 144,
                                       'count_geotag': 135,
                                       'max_id': 264152123716009985,
                                       'min_id': 263794534138384384},
          datetime.date(2012, 11, 2): {'count': 128,
                                       'count_geotag': 114,
                                       'max_id': 264516696830787584,
                                       'min_id': 264156297061089280},
          datetime.date(2012, 11, 3): {'count': 142,
                                       'count_geotag': 123,
                                       'max_id': 264877088203497472,
                                       'min_id': 264520266154143744},
          datetime.date(2012, 11, 4): {'count': 145,
                                       'count_geotag': 120,
                                       'max_id': 265240323310104577,
                                       'min_id': 264880737247952896},
          datetime.date(2012, 11, 5): {'count': 70,
                                       'count_geotag': 59,
                                       'max_id': 265395911016984576,
                                       'min_id': 265243192201474049}}}
